From f275207c1c5b29cfb115922a326c74b8af36489f Mon Sep 17 00:00:00 2001
From: srcnert <sercanerat@gmail.com>
Date: Mon, 25 May 2026 17:25:50 +0300
Subject: [PATCH 2/3] 0002-Fixing-lora-sleep-problem

---
 drivers/lora/loramac-node/sx126x.c | 19 +++++++++++++++++++
 1 file changed, 19 insertions(+)

diff --git a/drivers/lora/loramac-node/sx126x.c b/drivers/lora/loramac-node/sx126x.c
index a1de81b0a68..9a785ad9b15 100644
--- a/drivers/lora/loramac-node/sx126x.c
+++ b/drivers/lora/loramac-node/sx126x.c
@@ -296,6 +296,15 @@ void SX126xSetOperatingMode(RadioOperatingModes_t mode)
 		break;
 
 	case MODE_SLEEP:
+#if HAVE_GPIO_ANTENNA_ENABLE
+		gpio_pin_configure_dt(&dev_config.antenna_enable, GPIO_DISCONNECTED);
+#endif
+#if HAVE_GPIO_TX_ENABLE
+		gpio_pin_configure_dt(&dev_config.tx_enable, GPIO_DISCONNECTED);
+#endif
+#if HAVE_GPIO_RX_ENABLE
+		gpio_pin_configure_dt(&dev_config.rx_enable, GPIO_DISCONNECTED);
+#endif
 		/* Additionally disable the DIO1 interrupt to save power */
 		sx126x_dio1_irq_disable(&dev_data);
 		__fallthrough;
@@ -373,6 +382,16 @@ void SX126xWakeup(void)
 {
 	int ret;
 
+#if HAVE_GPIO_ANTENNA_ENABLE
+	gpio_pin_configure_dt(&dev_config.antenna_enable, GPIO_OUTPUT_INACTIVE);
+#endif
+#if HAVE_GPIO_TX_ENABLE
+	gpio_pin_configure_dt(&dev_config.tx_enable, GPIO_OUTPUT_INACTIVE);
+#endif
+#if HAVE_GPIO_RX_ENABLE
+	gpio_pin_configure_dt(&dev_config.rx_enable, GPIO_OUTPUT_INACTIVE);
+#endif
+
 	/* Reenable DIO1 when waking up */
 	sx126x_dio1_irq_enable(&dev_data);
 
-- 
2.52.0

